module/epochs/epoch_lookup.go

```diff
@@ -28,10 +28,38 @@ func NewEpochLookup(state protocol.State) *EpochLookup {
 // (even if the node does happen to have that stored in the underlying storage)
 // -- these queries indicate a bug in the querier.
 func (l *EpochLookup) EpochForView(view uint64) (epochCounter uint64, err error) {
-	previous := l.state.Final().Epochs().Previous()
-	current := l.state.Final().Epochs().Current()
-	next := l.state.Final().Epochs().Next()
+	epochs := l.state.Final().Epochs()
+	previous := epochs.Previous()
+	current := epochs.Current()
+	next := epochs.Next()
 
+	// TMP: EMERGENCY EPOCH CHAIN CONTINUATION [EECC]
+	//
+	// If the given view is within the bounds of the next epoch, and the epoch
+	// has not been set up or committed, we pretend that we are still in the
+	// current epoch and return that epoch's counter.
+	//
+	// This is used to determine which Random Beacon key we will use to sign and
+	// verify blocks and votes. The failure case we are avoiding here is if the
+	// DKG for next epoch failed and there is no Random Beacon key for that epoch,
+	// or if the next epoch failed for any other reason. In either case we will
+	// continue using the last valid Random Beacon key until the next spork.
+	//
+	currentFinalView, err := current.FinalView()
+	if err != nil {
+		return 0, err
+	}
+	if view > currentFinalView {
+		_, err := next.DKG() // either of the following errors indicates that we have transitioned into EECC
+		if errors.Is(err, protocol.ErrEpochNotCommitted) || errors.Is(err, protocol.ErrNextEpochNotSetup) {
+			return current.Counter()
```

Comment:
I would suggest to return the error, and make another function like `EpochForSignerView` for [the SignerStore to call](https://github.com/onflow/flow-go/blob/9706b4ca8057bad3ad241b44818077e898d6aca6/module/signature/signer_store.go#L34). 

Because different callers will expect to return different epoch counter for views that have no epoch committed:
- the signer store expects to return the current epoch
- the leader selection expects to returns the next epoch

Reply:
👍  Added in https://github.com/onflow/flow-go/pull/1250/commits/7e85f17d56e54303b14bd7a63a1877f9a74fba0d